Release/0.0.1

thomas-lefloch opened the pull request.

cesarGattano committed:
feat: :sparkles: Création des fonctions create_*_in_db dans crud/ pour les 4 tables

cesarGattano committed:
feat: :sparkles: Population de la base de données pour toutes les tables avec des données aléatoires.

thomas-lefloch committed:
feat: roles
Les utilisateurs ont maintenant un champ role
Ajout d'un script pour inserer un compte admin et les roles de bases (admin, staff, client)
Ajout de la route GET /roles

thomas-lefloch committed:
Merge pull request #13 from cesarGattano/menu_item_id_routes

Menu item id routes

thomas-lefloch committed:
feat: route POST /login
Authentification Bear Token. Renvoi un JWT avec de l'utilisateur

cesarGattano committed:
Delete empty file 3

cesarGattano committed:
Merge pull request #14 from cesarGattano/feature/order

Création des routes get /by_date

cesarGattano committed:
Merge branch 'dev' into menu_item_id_routes

thomas-lefloch committed:
feat: login
Permet de demander un access_token
Une route test pour restreindre l'acces

cesarGattano committed:
feat: :sparkles: hachage et salage des mots de passe dans user.py

cesarGattano committed:
feat: :sparkles: hachage et salage du mot de passe dans setup_db.py

cesarGattano committed:
feat: :sparkles: hachage et salage des mots de passe dans login.py. A TESTER

JawadBerrhili committed:
Merge pull request #15 from cesarGattano/menu_item_id_routes

Mise en base de données aléatoires pour les tests

thomas-lefloch committed:
feat: création et modification des roles dans les routes users

JawadBerrhili committed:
feat: :construction: Ajout des routes pour recuperer une commande avec ses details et de la route admin pour recuperer toutes les commandes"

JawadBerrhili committed:
fix: Nettoyage et modification de la route /by_client/

cesarGattano committed:
feat: :sparkles: Ajout du sel comme champ de la table user_info

thomas-lefloch committed:
feat: gestions des roles pour les routes /users

thomas-lefloch committed:
Merge branch 'feature/users' of github.com:cesarGattano/restau_simplon into feature/users

cesarGattano committed:
feat: :sparkles: Nouvelle version alembic avec ajout du champ sel dans la base user_info

cesarGattano committed:
feat: :sparkles: Mise à jour setup_db.py pour inclure le sel pour les utilisateurs test

cesarGattano committed:
feat: :sparkles: Gestion du salage par user dans les routes users et logins

thomas-lefloch committed:
feat: ressources non trouvées (404)

thomas-lefloch committed:
Merge branch 'feature/users' of github.com:cesarGattano/restau_simplon into feature/users

cesarGattano committed:
Merge pull request #16 from cesarGattano/feature/order

Ajout des routes /by_order et /with_details pour admin. Amelioration de la doc

thomas-lefloch committed:
fix: comparaison mdp et hash

JawadBerrhili committed:
feat: :sparkles: Ajout des relationship avec back_populate et optimisation des requetes

thomas-lefloch committed:
feat: gestion de l'unicité des emails

cesarGattano committed:
Merge pull request #18 from cesarGattano/feature/add-relationship

feat: :sparkles: Ajout des relationship avec back_populate

thomas-lefloch committed:
fix: roles optionels

thomas-lefloch committed:
Merge branch 'dev' of github.com:cesarGattano/restau_simplon into feature/users

thomas-lefloch committed:
fix: suppression code inutile

cesarGattano committed:
Merge pull request #19 from cesarGattano/feature/users

Feature/users

cesarGattano committed:
feat: :sparkles: Nullable order_id dans le modèle OrderDetail

cesarGattano committed:
fix: :bug: Réparation de database.py suite à la mise en place des relations de table

user_info avec role et order avec order_detail

cesarGattano committed:
feat: :wastebasket: Suppression de setup_db.py

thomas-lefloch committed:
Merge pull request #20 from cesarGattano/fix_db_init

Fix db init

thomas-lefloch committed:
feat: cors sur localhost:8000

Ne fonctionne pas ! je ne sais pas pourquoi. On verra plus tard

thomas-lefloch committed:
feat: une route /users/client pour la création de compte client

thomas-lefloch committed:
Merge branch 'dev' of github.com:cesarGattano/restau_simplon into feature/creation-compte-client

thomas-lefloch committed:
fix: changement non voulu

cesarGattano committed:
Merge pull request #21 from cesarGattano/feature/creation-compte-client

Feature/creation compte client

cesarGattano committed:
feat: :sparkles: Ajout des permissions pour les routes menu_items

JawadBerrhili committed:
feat: :sparkles: Ajout des verifcation et restriction en fonction des roles pour les routes order

JawadBerrhili committed:
feat: :sparkles: Ajout des verifcation et restriction en fonction des roles pour les routes order

JawadBerrhili committed:
Merge pull request #22 from cesarGattano/feat/permission_menu_items

feat: :sparkles: Ajout des permissions pour les routes menu_items

JawadBerrhili committed:
Merge branch 'feature/order' of https://github.com/cesarGattano/restau_simplon into feature/order

thomas-lefloch committed:
Merge pull request #23 from cesarGattano/feature/order

feat: :sparkles: Ajout des verifcation et restriction en fonction des…

thomas-lefloch committed:
feat: pip freeze

JawadBerrhili committed:
Merge pull request #24 from cesarGattano/feature/pip-freeze

feat: pip freeze

cesarGattano reviewed (changes requested):
Il manque le fichier README

thomas-lefloch committed:
feat: ajout de la variable d'environnement DB_URI

thomas-lefloch committed:
feat: ajout des variables d'environnement liés au JWT
JWT_ALGORITHM
JWT_SECRET
ACCESS_TOKEN_EXPIRE_MINUTES

thomas-lefloch committed:
feat: variables environnements pour l'admin par défaut

cesarGattano committed:
Merge pull request #26 from cesarGattano/feat/env-variables

feature/env variables

thomas-lefloch committed:
docs: commandes docker

thomas-lefloch committed:
feat: dockerfile

thomas-lefloch committed:
feat: dockerfile

thomas-lefloch committed:
docs: commande lié au registry github

thomas-lefloch committed:
Merge branch 'feature/docker' of github.com:cesarGattano/restau_simplon into feature/docker

cesarGattano committed:
docs: ajout de *.env dans .gitignore et .dockerignore

cesarGattano committed:
feat: création compose.test.yaml

avec le container restausimplon-db + fichier environnement .db.env.example

cesarGattano committed:
docs: docker.md complété avec "lancer docker compose"

JawadBerrhili committed:
fix: :fire: Modification du fichier compose et suppression des db.env *

JawadBerrhili committed:
fix: Correction des fichier env

thomas-lefloch committed:
feat: Dockerfile pour la migration

thomas-lefloch committed:
Merge branch 'feature/docker' of github.com:cesarGattano/restau_simplon into feature/docker

thomas-lefloch committed:
Merge branch 'feature/docker' of github.com:cesarGattano/restau_simplon into feature/docker

thomas-lefloch committed:
docs: build en spécifiant un dockerfile

thomas-lefloch committed:
feat: Migration de la base
Elle est effectué après le healthcheck  de la base de donnée.

thomas-lefloch committed:
feat: insertion des données de test

thomas-lefloch committed:
fix: structure de fichier dans le conteneur

JawadBerrhili committed:
feat: Automatisation de l'API

JawadBerrhili committed:
docs: ajout des valeurs par défault dans le env.example pour l'api

thomas-lefloch committed:
feat: éxecution de l'application avec des droits limités

JawadBerrhili committed:
docs: ajout du fichier arbo.md pour avoir une arborescence

thomas-lefloch committed:
feat: environnement de prod
Accompagné de quelques fixs

thomas-lefloch committed:
Merge branch 'feature/docker' of github.com:cesarGattano/restau_simplon into feature/docker

thomas-lefloch committed:
Merge branch 'dev' of github.com:cesarGattano/restau_simplon into feature/docker

JawadBerrhili committed:
Merge pull request #27 from restausimplon-dreamteam5/feature/docker

Feature/docker

cesarGattano committed:
docs: première version README.md

thomas-lefloch committed:
docs: Ajout de plusieurs sections
- Variables d'environnement
- Construire la base de donnée
- Insérer les données de tests

cesarGattano committed:
docs: emojis ;)

cesarGattano committed:
docs: :memo: Mise à jour de requirements.txt avec pytest, pytest-cov

cesarGattano committed:
feat: :sparkles: Un premier test unitaire dans tests/schemas/test_menu_items.py

cesarGattano committed:
refactor: :building_construction: Création du sous-dosier units dans tests pour les tests unitaires

cesarGattano committed:
feat: :sparkles: Ajout de la fixture correct_menu_item_example dans tests/units/schemacs/test_menu_items.py

cesarGattano committed:
feat: :sparkles: Ajout de la fixture correct_menu_item_example dans tests/units/schemacs/test_menu_items.py

JawadBerrhili committed:
feat: :sparkles: Un test unitaire dans tests/schemas/test_order_detail.py

thomas-lefloch committed:
feat: tests pour le prénom

thomas-lefloch committed:
feat: tests pour le prénom

JawadBerrhili committed:
feat: :sparkles: Un test unitaire dans tests/schemas/test_order.py

cesarGattano committed:
docs: :memo: Mise à jour de .gitignore pour ignorer .vscode

cesarGattano committed:
docs: :memo: Mise à jour de .gitignore pour ignorer .vscode

cesarGattano committed:
feat: :sparkles: Ajout du unit test test_MenuItemCreate_init_too_long_name dans tests/units/schemas/menu_items.py

cesarGattano committed:
feat: :sparkles: Ajout du unit test test_MenuItemCreate_init_too_long_name dans tests/units/schemas/menu_items.py

cesarGattano committed:
feat: :sparkles: Paramétrisation des tests tests/units/schemas/menu_items.py pour s'appliquer à MenuItemCreate et MenuItemUpdate

cesarGattano committed:
feat: :sparkles: Paramétrisation des tests tests/units/schemas/menu_items.py pour s'appliquer à MenuItemCreate et MenuItemUpdate

cesarGattano committed:
feat: :sparkles: Paramétrisation des tests tests/units/schemas/menu_item.py pour accepter le schéma MenuItemOut

cesarGattano committed:
feat: :sparkles: Paramétrisation des tests tests/units/schemas/menu_item.py pour accepter le schéma MenuItemOut

thomas-lefloch committed:
Merge pull request #28 from restausimplon-dreamteam5/feature/tests-order

Feature/tests order

JawadBerrhili committed:
Merge pull request #29 from restausimplon-dreamteam5/feature/tests-users

Feature/tests users

thomas-lefloch committed:
Merge pull request #30 from restausimplon-dreamteam5/feature/tests_menu_items

Feature/tests menu items

cesarGattano committed:
feat: :sparkles: Test workflow "salutation"

cesarGattano committed:
Merge branch 'feature/tests' into feature/ci_flake8

thomas-lefloch committed:
ci: mypy

thomas-lefloch committed:
ci: mypy

thomas-lefloch committed:
ci: avec des saut de lignes

thomas-lefloch committed:
ci: avec des saut de lignes

JawadBerrhili committed:
feat: :sparkles: ajoute d'un test simple d’OrderCreate avec fixture

JawadBerrhili committed:
feat: :sparkles: ajoute d'un test simple d’OrderCreate avec fixture

JawadBerrhili committed:
feat: :sparkles: ajout d'un test OrderDetailCreate pour vérifier que la quantité est sup à 0

JawadBerrhili committed:
feat: :sparkles: ajout d'un test OrderDetailCreate pour vérifier que la quantité est sup à 0

cesarGattano committed:
Job flake8 dans le workflow

thomas-lefloch committed:
ci: fix syntax

thomas-lefloch committed:
ci: fix syntax

thomas-lefloch committed:
ci: uses -> run

thomas-lefloch committed:
ci: uses -> run

thomas-lefloch committed:
ci:  test en installant que mypy

thomas-lefloch committed:
ci:  test en installant que mypy

JawadBerrhili committed:
feat: :sparkles: ci(isort): ajout GitHub Actions workflow

JawadBerrhili committed:
feat: :sparkles: ci(isort): ajout GitHub Actions workflow

thomas-lefloch committed:
ci: changment de l'image à ubuntu
nécessaire pour lancer la pipeline sur github gratuitement

thomas-lefloch committed:
ci: changment de l'image à ubuntu
nécessaire pour lancer la pipeline sur github gratuitement

thomas-lefloch committed:
fix: typo

thomas-lefloch committed:
fix: typo

cesarGattano committed:
style: Mise à jour CI job flake8

thomas-lefloch committed:
fix: installation des dépendances

thomas-lefloch committed:
fix: installation des dépendances

thomas-lefloch committed:
ci: installation de toutes les dépendances

thomas-lefloch committed:
ci: installation de toutes les dépendances

JawadBerrhili committed:
style: trier les imports avec isort

JawadBerrhili committed:
style: trier les imports avec isort

thomas-lefloch committed:
Merge pull request #31 from restausimplon-dreamteam5/feature/tests-order

Feature/tests order

JawadBerrhili committed:
black: formater le code avec black

JawadBerrhili committed:
black: formater le code avec black

JawadBerrhili committed:
black: formatage du code ok

JawadBerrhili committed:
black: formatage du code ok

JawadBerrhili committed:
trier les imports avec isort + formatage avec black

JawadBerrhili committed:
trier les imports avec isort + formatage avec black

JawadBerrhili committed:
ci: aligner isort sur le profile black

JawadBerrhili committed:
ci: aligner isort sur le profile black

cesarGattano committed:
style: :art: Flake8 satisfait

cesarGattano committed:
Merge pull request #32 from restausimplon-dreamteam5/feature/tests-order

Feature/tests order

thomas-lefloch committed:
fix: rendre heureux mypy

thomas-lefloch committed:
fix: rendre heureux mypy

cesarGattano committed:
Merge branch 'feature/tests' into feature/ci_flake8

cesarGattano committed:
Flake8 ignore E501,W503 dans la CI

thomas-lefloch committed:
Merge branch 'feature/ci-mypy' of github.com:cesarGattano/restau_simplon into feature/ci-mypy

cesarGattano committed:
style: :art: isort/black satisfait

thomas-lefloch committed:
Merge pull request #33 from restausimplon-dreamteam5/feature/ci-mypy

Feature/ci mypy

cesarGattano committed:
feat: :green_heart: Flake8 ignore les dossiers .venv, migrations, .pytest_cache

cesarGattano committed:
style: :art: isort, flake8 satisfaits sur le dossier tests

cesarGattano committed:
Merge branch 'feature/tests' into feature/ci_flake8

cesarGattano committed:
style: :art: flake8 satisfait

cesarGattano committed:
style: :art: isort enfin satisfait !!!

thomas-lefloch committed:
Merge pull request #34 from restausimplon-dreamteam5/feature/ci_flake8

Feature/ci flake8

cesarGattano committed:
refactor: :green_heart: Pipeline qualité de code pour la CI

cesarGattano committed:
ci: :green_heart: CI unique, exécution des tests unitaires (push)

cesarGattano committed:
ci: :green_heart: Fixe l'éxecution de la CI

cesarGattano committed:
fix: :green_heart: export PYTHONPATH

cesarGattano committed:
test: test

cesarGattano committed:
test: test2

cesarGattano committed:
test: test3

cesarGattano committed:
test: test4

cesarGattano committed:
test: test5

JawadBerrhili committed:
test: test pytest.ini

cesarGattano committed:
test: test6

cesarGattano committed:
test: test7

cesarGattano committed:
style: :wastebasket: Nettoyage CI

JawadBerrhili committed:
test: test avec ajout pytest.ini

cesarGattano committed:
ci: :green_heart: Mode verbose pour les tests + max fail à 1

cesarGattano committed:
ci: :sparkles: Mise en place de la CI "tests d'intégration"

cesarGattano committed:
feat: :sparkles: Docker compose passe en mode build des images (pour la CI)

cesarGattano committed:
ci: :green_heart: Mise en place docker compose (tentative 1)

cesarGattano committed:
ci: :green_heart: Mise en place de la CI test d'intégration (tentative 2)

cesarGattano committed:
ci: :green_heart: Mise en place de la CI test d'intégration (tentative 3)

cesarGattano committed:
ci: :green_heart: Debbugage 1

cesarGattano committed:
ci: :green_heart: Debbugage 2

cesarGattano committed:
ci: :green_heart: Debuggage 3

cesarGattano committed:
ci: :green_heart: Debbugage 4

cesarGattano committed:
ci: :green_heart: Debbugage 5

cesarGattano committed:
ci: :green_heart: Debbugage 6

thomas-lefloch committed:
tests: premier test integration

thomas-lefloch committed:
ci: tests d'integration dans un conteneur séparé

thomas-lefloch committed:
fix(ci): premier test d'integration

thomas-lefloch committed:
fix(ci): test intégrations

thomas-lefloch committed:
ci: exclusion de la regle E402 de flake8

thomas-lefloch committed:
fix: nom de module en doublon

thomas-lefloch committed:
ci: fix "boucle" infini

thomas-lefloch committed:
ci: tests integration à part

thomas-lefloch committed:
fix(ci): environement API_URL

thomas-lefloch committed:
fix(ci): utilsationdu compose au lieu de l'image

thomas-lefloch committed:
fix(ci): dependances

thomas-lefloch committed:
ci: test

thomas-lefloch committed:
ci: test reconnection au réseau

thomas-lefloch committed:
ci: typo

thomas-lefloch committed:
ci: tout en un

thomas-lefloch committed:
ci: ??

thomas-lefloch committed:
ci: test

thomas-lefloch committed:
fix: typo

thomas-lefloch committed:
ci: ça devrait marcher !

thomas-lefloch committed:
ci: nom du job

cesarGattano committed:
test: :white_check_mark: Premiers tests intégration articles de menu

cesarGattano committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/tests

thomas-lefloch committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/tests

thomas-lefloch committed:
ci: test

cesarGattano committed:
ci: :green_heart: Mypy satisfait

cesarGattano committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/tests

thomas-lefloch committed:
ci: démarrage des services

thomas-lefloch committed:
ci: ne pas executer les services test par défault

cesarGattano committed:
test: :white_check_mark: variable environnement API_URL dans les test d'intégration

thomas-lefloch committed:
ci: meilleur noms d'étapes

thomas-lefloch committed:
ci: test qui échoue

thomas-lefloch committed:
fix: suppresion du test qui echoue

cesarGattano committed:
ci: :green_heart: Ajout du profile data

cesarGattano committed:
test: :white_check_mark: Test routes login et menu_items (par nom uniquement)

thomas-lefloch committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/exploration-ci

cesarGattano committed:
docs: :memo: Ignore les fichiers not_working*

thomas-lefloch committed:
fix(ci): installation de pytest

thomas-lefloch committed:
fix: appaisement de la ci

thomas-lefloch committed:
fix: ajout de requests

thomas-lefloch committed:
fix: types pour requests

thomas-lefloch committed:
Merge pull request #36 from restausimplon-dreamteam5/feature/exploration-ci

Feature/exploration ci

cesarGattano committed:
style: :art: isort satisfait

cesarGattano committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/tests

cesarGattano committed:
Merge branch 'feature/tests' into feature/test_author

JawadBerrhili committed:
fix: modification du la variable API_URL

cesarGattano committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/tests

cesarGattano committed:
Merge branch 'feature/tests' into feature/test_author

cesarGattano committed:
fix: :white_check_mark: Mot de passe admin correct

JawadBerrhili committed:
fix: modification du mot de passe ADMIN_PASSWORD

cesarGattano committed:
Merge branch 'feature/tests' of github.com:cesarGattano/restau_simplon into feature/test_author

cesarGattano committed:
Merge pull request #37 from restausimplon-dreamteam5/feature/test_author

Feature/test author

thomas-lefloch committed:
Merge pull request #35 from restausimplon-dreamteam5/feature/tests

Feature/tests

thomas-lefloch committed:
Merge branch 'dev' of github.com:cesarGattano/restau_simplon into feature/readme

cesarGattano committed:
docs: :memo: Docstrings périmètre article de menu

cesarGattano committed:
style: :art: flake8 satisfait

JawadBerrhili committed:
docs: :fire: Ajout docstring aux schemas, routes

thomas-lefloch committed:
docs: docstrings des routes

JawadBerrhili committed:
Merge pull request #38 from restausimplon-dreamteam5/feature/menu_item_docstrings

Feature/menu item docstrings

thomas-lefloch committed:
docs: descriptions des schémas users

thomas-lefloch committed:
Merge pull request #39 from restausimplon-dreamteam5/feature/orders-documentation

docs: :fire: Ajout docstring aux schemas, routes pour order

thomas-lefloch committed:
Merge branch 'dev' of github.com:cesarGattano/restau_simplon into feature/user-docstrings

cesarGattano committed:
docs: :memo: Mise à jour README migrations

JawadBerrhili committed:
Merge pull request #40 from restausimplon-dreamteam5/feature/user-docstrings

Feature/user docstrings

cesarGattano committed:
docs: :memo: Suppression arbo.md

cesarGattano committed:
docs: :memo: Mise à jour README général

thomas-lefloch committed:
docs: CI/CD & tests

cesarGattano committed:
docs: :memo: typos

cesarGattano committed:
Merge pull request #41 from restausimplon-dreamteam5/feature/readme

Feature/readme
